src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -703,17 +704,28 @@ def aggregate(
             # native pandas raise error with message "no result", here we raise a more readable error.
             raise ValueError("No column to aggregate on.")
 
-        func = validate_and_try_convert_agg_func_arg_func_to_str(
-            agg_func=func,
-            obj=self,
-            allow_duplication=False,
-            axis=axis,
-        )
+        if func is None:
+            if axis == 1:
+                raise ValueError(
+                    "`func` must not be `None` when `axis=1`. Named aggregations are not supported with `axis=1`."
```

Comment:
This error isn't the same - groupby_agg isn't supported (at all) when axis=1 by us, while neither we nor pandas support df.agg with axis=1 and named aggregations - I'm not sure it makes sense to move the error, since we'd still need the error checking in groupby_agg anyways.

Reply:
I double checked, in pandas 2.0, groupby with axis = 1 is not supported anymore, i got error  NotImplementedError: axis other than 0 is not supported

I think we can keep this special for agg

Reply:
However, since we intended to put those functionality support decision for the particular backend, can we move this check to the query compiler agg to have it at the same place with the other not implemented check

Reply:
This is a pandas error message parity check - so shouldn't it be in the frontend? For context, pandas does one of two things for named agg with axis=1:
1. If we call `agg` with index values that exist, we get a KeyError: col_name
2. If we call `agg` with index values that do not exist, we get a Columns do not exist error.
I've opened an issue [here](https://github.com/pandas-dev/pandas/issues/58807) for the pandas folks to standardize the error message, but based off of that, it seems that df.agg(axis=1, **named_aggs) is unsupported as a feature by pandas. Therefore, this isn't a backend specific decision, and should remain at the front end?

Reply:
That is fine, I think i was mis-reading the error, i thought it is a case that is not intended to support by us, which is suitable to decide at the query compiler with other not implemented error. However, that seems something that is not intended to support anyway, which should be fine at frontend layer.

Reply:
Sounds good! Marking as resolved!